build.sh

```diff
@@ -281,11 +281,11 @@ build_native()
             if [[ $__StaticAnalyzer == 1 ]]; then
                 scan_build=scan-build
             fi
-            echo "Invoking \"$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh\" \"$__ProjectRoot\" $__ClangMajorVersion \"$__ClangMinorVersion\" $platformArch $__BuildType $__CodeCoverage $scan_build $generator $extraCmakeArguments $__cmakeargs"
-            source "$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh" "$__ProjectRoot" $__ClangMajorVersion "$__ClangMinorVersion" $platformArch $__BuildType $__CodeCoverage $scan_build $generator "$extraCmakeArguments" "$__cmakeargs"
+            echo "Invoking \"$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh\" \"$__ProjectRoot\" $__ClangMajorVersion \"$__ClangMinorVersion\" $platformArch "$__ProjectRoot/src/pal/tools" $__BuildType $__CodeCoverage $scan_build $generator $extraCmakeArguments $__cmakeargs"
```

Comment:
Could you please extract the script dir to a variable and then use it in all of these commands (including in the paths of the .sh files)?